src/Umbraco.Infrastructure/Services/CacheInstructionService.cs

```diff
@@ -185,8 +185,18 @@ public void DeliverInstructionsInBatches(IEnumerable<RefreshInstruction> instruc
                 }
             }
 
-            private CacheInstruction CreateCacheInstruction(IEnumerable<RefreshInstruction> instructions, string localIdentity) =>
-                new(0, DateTime.UtcNow, JsonConvert.SerializeObject(instructions, Formatting.None), localIdentity, instructions.Sum(x => x.JsonIdCount));
+            private CacheInstruction CreateCacheInstruction(IEnumerable<RefreshInstruction> instructions, string localIdentity)
+                => new(
+                    0,
+                    DateTime.UtcNow,
+                    JsonConvert.SerializeObject(instructions, new JsonSerializerSettings()
+                    {
+                        Formatting = Formatting.None,
+                        DefaultValueHandling = DefaultValueHandling.Ignore,
+                        NullValueHandling = NullValueHandling.Ignore,
+                    }),
+                    localIdentity,
+                    instructions.Sum(x => x.JsonIdCount));
```

Comment:
This simple change results in way less JSON data stored in the `umbracoCacheInstruction` table, as the `jsonInstruction` column previously contained:
```json
[{"RefreshType":3,"RefresherId":"e057af6d-2ee6-41f4-8045-3694010f0aa6","GuidId":"00000000-0000-0000-0000-000000000000","IntId":0,"JsonIds":"[-1]","JsonIdCount":1,"JsonPayload":null},{"RefreshType":3,"RefresherId":"e057af6d-2ee6-41f4-8045-3694010f0aa6","GuidId":"00000000-0000-0000-0000-000000000000","IntId":0,"JsonIds":"[-1]","JsonIdCount":1,"JsonPayload":null}]
```

And now only the actual data:
```json
[{"RefreshType":3,"RefresherId":"e057af6d-2ee6-41f4-8045-3694010f0aa6","JsonIds":"[-1]","JsonIdCount":1},{"RefreshType":3,"RefresherId":"e057af6d-2ee6-41f4-8045-3694010f0aa6","JsonIds":"[-1]","JsonIdCount":1}]
```

It's still quite verbose (long property names and counts that can be calculated), but changing that would break compatibility...

Reply:
@LennardF1989 suggested to make `JsonIdCount` a faux property, but while looking into it a bit more, it's only used to calculate the instruction count when saving it to the database (the `Sum()` in the above code). So that means it doesn't need to be in the serialised JSON at all... I've added attributes to only serialize the required properties, making the instructions JSON a little smaller again 😄